backend/src/graphql/resolver/TransactionResolver.ts

```diff
@@ -120,142 +76,70 @@ export class TransactionResolver {
-
-        const otherUser = userIndiced.find((u) => u.id === userTransaction.linkedUserId)
-        switch (userTransaction.typeId) {
-          case TransactionTypeId.CREATION:
-            finalTransaction.name = 'Gradido Akademie'
-            finalTransaction.type = TransactionType.CREATION
-            break
-          case TransactionTypeId.SEND:
-            finalTransaction.type = TransactionType.SEND
-            if (otherUser) {
-              finalTransaction.name = otherUser.firstName + ' ' + otherUser.lastName
-              finalTransaction.email = otherUser.email
-            }
-            break
-          case TransactionTypeId.RECEIVE:
-            finalTransaction.type = TransactionType.RECIEVE
-            if (otherUser) {
-              finalTransaction.name = otherUser.firstName + ' ' + otherUser.lastName
-              finalTransaction.email = otherUser.email
-            }
-            break
-          default:
-            throw new Error('invalid transaction')
-        }
-        if (i > 0 || !skipFirstTransaction) {
-          transactions.push(finalTransaction)
-        }
-
-        if (i === userTransactions.length - 1 && decay) {
-          const now = new Date()
-          const decay = calculateDecay(
-            Number(userTransaction.balance),
-            userTransaction.balanceDate,
-            now,
-          )
-          const balance = Number(userTransaction.balance) - decay.balance
-
-          const decayTransaction = new Transaction()
-          decayTransaction.type = 'decay'
-          decayTransaction.balance = roundCeilFrom4(balance)
-          decayTransaction.decayDuration = decay.decayDuration
-          decayTransaction.decayStart = decay.decayStart
-          decayTransaction.decayEnd = decay.decayEnd
-          transactions.push(decayTransaction)
-        }
+    // find involved users
+    let involvedUserIds: number[] = []
+    userTransactions.forEach((transaction: dbTransaction) => {
+      involvedUserIds.push(transaction.userId)
+      if (transaction.linkedUserId) {
+        involvedUserIds.push(transaction.linkedUserId)
       }
+    })
+    // remove duplicates
+    involvedUserIds = involvedUserIds.filter((value, index, self) => self.indexOf(value) === index)
+    // We need to show the name for deleted users for old transactions
+    const involvedDbUsers = await dbUser
+      .createQueryBuilder()
+      .withDeleted()
+      .where('id IN (:...userIds)', { userIds: involvedUserIds })
+      .getMany()
+    const involvedUsers = involvedDbUsers.map((u) => new User(u))
+
+    const self = new User(user)
+    const transactions: Transaction[] = []
 
-      if (order === Order.DESC) {
-        transactions.reverse()
-      }
+    // decay transaction
+    if (currentPage === 1 && order === Order.DESC) {
+      const now = new Date()
+      transactions.push(
+        virtualDecayTransaction(lastTransaction.balance, lastTransaction.balanceDate, now, self),
```

Comment:
lastTransaction my be null for new users

Reply:
they will have zero transactions - cool?